app/controllers/courses_controller.rb

```diff
@@ -0,0 +1,89 @@
+class CoursesController < ApplicationController
+  before_action :set_course, only: [:show, :edit, :update, :destroy]
+  before_action :authenticate_user!
+  before_action :check_permitted_or_not , only: [:edit , :destroy]
+
+  def index
+    @courses = Course.all
+  end
+
+  def show; end
+
+  def check_permitted_manage_courses_or_not
+    if current_user && current_user.usertype == 1
+      redirect_to courses_path
+    end
+  end
+
+  def new
+    @course = Course.new
+  end
+
+  def edit; end
+
+  def create
+    @course = Course.new　course_params
+    @course.user_id = current_user.id
+
+    respond_to do |format|
+      if @course.save
+        format.html { redirect_to @course, notice: t("controller.course.create") }
+        format.json { render :show, status: :created, location: @course }
+      else
+        format.html { render :new }
+        format.json { render json: @course.errors, status: :unprocessable_entity }
+      end
+    end
+  end 
+
+  def update
+    respond_to do |format|
+      if @course.update(course_params)
+        format.html { redirect_to @course, notice: t("controller.course.updated") }
+        format.json { render :show, status: :ok, location: @course }
+      else
+        format.html { render :edit }
+        format.json { render json: @course.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
+    @course.destroy
+    respond_to do |format|
+      format.html { redirect_to courses_path, notice: t("controller.course.updated") }
+      format.json { head :no_content }
+    end
+  end
+
+  def check_permitted_or_not
+    if current_user
+      if @course.try(:user).try(:id) != current_user.id
+        redirect_to courses_path()
+      end
+    end
+  end
+
+  def authenticate_user!
```

Comment:
Why don't you login and then go to this page

remove this function plz

Reply:
<img width="1015" alt="0925" src="https://user-images.githubusercontent.com/68420253/94239906-10ecc480-ff45-11ea-9d85-d3a3c19804da.png">
I get error for it

Reply:
> Why don't you login and then go to this page
> 
> remove this function plz

<img width="1015" alt="0925" src="https://user-images.githubusercontent.com/68420253/94240612-1a2a6100-ff46-11ea-9a83-37f1ba577e0d.png">

I got it error